src/viam/components/audio_input/service.py

```diff
@@ -0,0 +1,102 @@
+import wave
+from datetime import timedelta
+from io import BytesIO
+
+from google.api.httpbody_pb2 import HttpBody
+from grpclib import GRPCError, Status
+from grpclib.server import Stream
+
+from viam.components.service_base import ComponentServiceBase
+from viam.errors import ComponentNotFoundError, NotSupportedError
+from viam.gen.component.audioinput.v1.audioinput_pb2 import SampleFormat
+from viam.proto.component.audioinput import (
+    AudioInputServiceBase,
+    ChunksRequest,
+    ChunksResponse,
+    PropertiesRequest,
+    PropertiesResponse,
+    RecordRequest,
+)
+
+from .audio_input import AudioInput
+
+
+class AudioInputService(AudioInputServiceBase, ComponentServiceBase[AudioInput]):
+    """
+    gRPC Service for a generic AudioInput
+    """
+
+    RESOURCE_TYPE = AudioInput
+
+    async def Chunks(self, stream: Stream[ChunksRequest, ChunksResponse]) -> None:
+        request = await stream.recv_message()
+        assert request is not None
+        try:
+            audio_input = self.get_component(request.name)
+        except ComponentNotFoundError as e:
+            raise e.grpc_error
+
+        audio_stream = await audio_input.stream()
+        first_chunk = await audio_stream.__anext__()
+        await stream.send_message(ChunksResponse(info=first_chunk.info))
+        await stream.send_message(ChunksResponse(chunk=first_chunk.chunk))
+
+        async for audio in audio_stream:
+            await stream.send_message(ChunksResponse(chunk=audio.chunk))
+
+    async def Properties(self, stream: Stream[PropertiesRequest, PropertiesResponse]) -> None:
+        request = await stream.recv_message()
+        assert request is not None
+        try:
+            audio_input = self.get_component(request.name)
+        except ComponentNotFoundError as e:
+            raise e.grpc_error
+        response = (await audio_input.get_properties()).proto
+        await stream.send_message(response)
+
+    async def Record(self, stream: Stream[RecordRequest, HttpBody]) -> None:
+        raise NotSupportedError("Recording audio input is not supported").grpc_error
+
+        # TODO: Eventually implement recording
```

Comment:
probably should link this to a specific ticket.

Reply:
Talked with the product team about it -- this is more of a "if people ask for it" kind of reminder. Not something we need to keep track of (yet)